### PR TITLE
190 filter by incidence rate or count for ae tables

### PR DESCRIPTION
**Pull request by fukuhiro2023**
This PR ennables:
 - To filter an AE specific table by a specified criteria of count or proportion in `format_ae_specific()`.
 - To sort an AE specific table in alphabetical or count-based order with a specified column in `format_ae_specific()`.

Also, `tlf_ae_specific()` is updated to add a subtilte regarding filtering.

### CODE REVIEW

**Comment by fukuhiro2023** [comment]
I need to check further. Once completed, I will reopen this PR.

**Comment by fukuhiro2023** [comment]
It’s ready, please review the updates.

File: R/format_ae_specific.R [hunk 1]
```diff
@@ -35,6 +35,16 @@
...
 #'   - `dur`: Average of adverse event duration.
 #'   - `events`: Average number of adverse event per subject.
+#' @param filter_method A character value to specify whether rows will be filtered
+#'    based on participant count or percent incidence.
+#' @param filter_criteria A numeric value to display rows where at least
+#'    one therapy group has a percent incidence or participant count
+#'    greater than or equal to the specified value.
+#' @param sort_order A character value to specify sorting order:
+#'  - `alpha`: Sort by alphabetical order.
+#'  - `countdes`: Sort by count in descending order.
```
Comment (LittleBeannie) [inline R/format_ae_specific.R:45]: Since count descending are 2 words, can we call it count_desc?

File: R/format_ae_specific.R [hunk 2]
```diff
@@ -35,6 +35,16 @@
...
+#'  - `countasc`: Sort by count in ascending order.
```
Comment (LittleBeannie) [inline R/format_ae_specific.R:46]: Echo with my previous comment, can we call it count_asc?

File: R/format_ae_specific.R [hunk 3]
```diff
@@ -35,6 +35,16 @@
...
+#' @param sort_column A character value of `group` in `outdata` used to sort a table with.
 #' @param mock A boolean value to display mock table.
 #'
 #' @return A list of analysis raw datasets.
```
Comment (LittleBeannie) [inline R/format_ae_specific.R:50]: Can we add an example explaining the usage of filtering by incidence rate?

File: R/format_ae_specific.R [hunk 4]
```diff
@@ -214,21 +230,96 @@ format_ae_specific <- function(outdata,
...
+
+  # Additional options for AE specific
+  if ("prepare_ae_specific" %in% as.character(outdata$prepare_call)) {
+    if ("soc_name" %in% names(outdata)) {
+      soc_name <- outdata$soc_name
+    }
+
+    # Filtering by criteria
+    if (filter_criteria > 0) {
+      if (toupper(filter_method) == "PERCENT") {
```
Comment (LittleBeannie) [inline R/format_ae_specific.R:243]: If we have `match.arg()`, I guess we no longer need the `toupper()`? If you agree, then there is a similar issue in line 261, 280, 289.

File: R/format_ae_specific.R [hunk 5]
```diff
@@ -214,21 +230,96 @@ format_ae_specific <- function(outdata,
...
+      outdata$order <- outdata$order[filter_logic]
+      soc_name <- soc_name[filter_logic]
+    }
+
+    # Get index of sort column
+    if (toupper(sort_order) %in% c("COUNTDES", "COUNTASC")) {
+      index_group <- which(outdata$group == sort_column)
+      if (length(index_group) == 0) {
+        message(paste(
+          'If `sort_order` = "countdes" or "countasc", `sort_column` should be specified as an existing column name.',
```
Comment (LittleBeannie) [inline R/format_ae_specific.R:265]: I like this message!

File: R/tlf_ae_specific.R [hunk 6]
```diff
@@ -94,13 +94,35 @@ tlf_ae_specific <- function(outdata,
...
     )
+
+    filter_criteria_text <- paste0(
+      ifelse(outdata$filter_criteria > 0, "\\geq ", "> "),
+      outdata$filter_criteria
+    )
+    filter_method_text <- ifelse(outdata$filter_method == "percent", "%", "Participants with an Adverse Event")
+    filter_text <- paste("(Incidence", filter_criteria_text, filter_method_text)
+    filter_text <- ifelse(n_group > 1,
+      paste(filter_text, "in One or More Treatment Groups)"),
```
Comment (LittleBeannie) [inline R/tlf_ae_specific.R:113]: Li, here is the answer you are looking for.

**Review by LittleBeannie — COMMENTED** [review]
Thanks @fukuhiro2023 for all the work!!! I only have some minor/editorial comments.

**Review by LittleBeannie — APPROVED** [review]

**Pull request merged**